app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenter.java

```diff
@@ -20,6 +21,38 @@ public void onButtonParkingSpaceReservationPickerPressed(DatePickerDialog.OnDate
         view.showDatePickerDialog(listener);
     }
 
+    @Override
+    public void onButtonParkingSpaceReservationSavePressed() {
+        model.setParkingSpace(view.getParkingSpace());
+        model.setSecurityCode(view.getSecurityCode());
+        ReservationCheck reservationCheck = model.checkFields();
+        switch (reservationCheck) {
+            case MISSING_DATE_START:
+                view.showMissingDateStartToast();
+                break;
+            case MISSING_TIME_START:
+                view.showMissingTimeStartToast();
+                break;
+            case MISSING_DATE_END:
+                view.showMissingDateEndToast();
+                break;
+            case MISSING_TIME_END:
+                view.showMissingTimeEndToast();
+                break;
+            case MISSING_PARKING_SPACE:
+                view.showMissingParkingSpaceToast();
+                break;
+            case MISSING_SECURITY_CODE:
+                view.showMissingSecurityCodeToast();
+                break;
+            case SUCCESS:
+                model.makeReservation();
+                view.showSaveMessage();
+                view.backToActivityMain();
+                break;
```

Comment:
this validation will be develop in the next PR. Now we only include saving functionallity

Reply:
Deleted it!